DQMOffline/L1Trigger/src/L1TStage2CaloLayer2Offline.cc

```diff
@@ -0,0 +1,596 @@
+  recoMHTPhi = TVector2::Phi_mpi_pi(mht.Phi());
+
+  // if no reco value, relative resolution does not make sense -> sort to overflow
+  double outOfBounds = 9999;
+  double resolutionMET = recoMET > 0 ? (l1MET - recoMET) / recoMET : outOfBounds;
+  double resolutionMETPhi = abs(recoMETPhi) > 0 ? (l1METPhi - recoMETPhi) / recoMETPhi : outOfBounds;
+  double resolutionMHT = recoMHT > 0 ? (l1MHT - recoMHT) / recoMHT : outOfBounds;
+  double resolutionMHTPhi = abs(recoMHTPhi) > 0 ? (l1MHTPhi - recoMHTPhi) / recoMHTPhi : outOfBounds;
+  double resolutionETT = recoETT > 0 ? (l1ETT - recoETT) / recoETT : outOfBounds;
+  double resolutionHTT = recoHTT > 0 ? (l1HTT - recoHTT) / recoHTT : outOfBounds;
+
+  using namespace dqmoffline::l1t;
+  fill2DWithinLimits(h_L1METvsCaloMET_, recoMET, l1MET);
+  fill2DWithinLimits(h_L1MHTvsRecoMHT_, recoMHT, l1MHT);
+  fill2DWithinLimits(h_L1METTvsCaloETT_, recoETT, l1ETT);
+  fill2DWithinLimits(h_L1HTTvsRecoHTT_, recoHTT, l1HTT);
+
+  fill2DWithinLimits(h_L1METPhivsCaloMETPhi_, recoMETPhi, l1METPhi);
+  fill2DWithinLimits(h_L1MHTPhivsRecoMHTPhi_, recoMHTPhi, l1MHTPhi);
+
+  fillWithinLimits(h_resolutionMET_, resolutionMET);
+  fillWithinLimits(h_resolutionMHT_, resolutionMHT);
+  fillWithinLimits(h_resolutionETT_, resolutionETT);
+  fillWithinLimits(h_resolutionHTT_, resolutionHTT);
+
+  fillWithinLimits(h_resolutionMETPhi_, resolutionMETPhi);
+  fillWithinLimits(h_resolutionMHTPhi_, resolutionMHTPhi);
+
+  // efficiencies
+  for (auto threshold : metEfficiencyThresholds_) {
+    fillWithinLimits(h_efficiencyMET_total_[threshold], recoMET);
+    if (l1MET > threshold)
+      fillWithinLimits(h_efficiencyMET_pass_[threshold], recoMET);
+  }
+
+  for (auto threshold : mhtEfficiencyThresholds_) {
+    fillWithinLimits(h_efficiencyMHT_total_[threshold], recoMHT);
+    if (l1MHT > threshold)
+      fillWithinLimits(h_efficiencyMHT_pass_[threshold], recoMHT);
+  }
+
+  for (auto threshold : ettEfficiencyThresholds_) {
+    fillWithinLimits(h_efficiencyETT_total_[threshold], recoETT);
+    if (l1ETT > threshold)
+      fillWithinLimits(h_efficiencyETT_pass_[threshold], recoETT);
+  }
+
+  for (auto threshold : httEfficiencyThresholds_) {
+    fillWithinLimits(h_efficiencyHTT_total_[threshold], recoHTT);
+    if (l1HTT > threshold)
+      fillWithinLimits(h_efficiencyHTT_pass_[threshold], recoHTT);
+  }
+
+}
+
+void L1TStage2CaloLayer2Offline::fillJets(edm::Event const& e, const unsigned int nVertex)
+{
+  edm::Handle<l1t::JetBxCollection> l1Jets;
+  e.getByToken(stage2CaloLayer2JetToken_, l1Jets);
+
+  edm::Handle<reco::CaloJetCollection> caloJets;
+  e.getByToken(theCaloJetCollection_, caloJets);
+
+  if (!caloJets.isValid()) {
+    edm::LogError("L1TStage2CaloLayer2Offline") << "invalid collection: calo jets " << std::endl;
+    return;
+  }
+  if (!l1Jets.isValid()) {
+    edm::LogError("L1TStage2CaloLayer2Offline") << "invalid collection: L1 jets " << std::endl;
+    return;
+  }
+
+  if (caloJets->size() == 0) {
```

Comment:
Fair enough. These are only relevant for debugging I guess, will change to `edm::LogDebug`